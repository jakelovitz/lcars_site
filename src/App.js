import React from 'react';
import './assets/styling.css';
import ScrollToTopButton from './components/ScrollToTop';
import { Link } from 'react-router-dom';

function App({ children, footer }) {
  return (
    <div>
      <div className="wrap" id="gap">
        <div className="left-frame">
          <div>
            <div className="panel-3">
              <span className="hop"></span>
            </div>
            <div className="sidebar-buttons">
              <Link to="/" id="0">
                About Me
              </Link>
              <Link to="/testing" id="1">
                Testing
              </Link>
              <Link to="/library" id="2">
                Library
              </Link>
            </div>
            <div className="panel-4">
              <span className="hop"></span>
            </div>
            <div className="panel-5">
              <span className="hop"></span>
            </div>
            <div className="panel-6">
              <span className="hop"></span>
            </div>
            <div className="panel-7">
              <span className="hop"></span>
            </div>
            <div className="panel-8">
              <span className="hop"></span>
            </div>
            <div className="panel-9">
              <span className="hop"></span>
            </div>
          </div>
          <div className="panel-10">
            <span className="hop"></span>
          </div>
        </div>

        <div className="right-frame">
          <div className="bar-panel">
            <div className="bar-6"></div>
            <div className="bar-7"></div>
            <div className="bar-8"></div>
            <div className="bar-9">
              <div className="bar-9-inside"></div>
            </div>
          </div>

          <div className="runner">
            <div className="corner-bg">
              <div className="corner"></div>
            </div>
            <div className="block-2"></div>
            <div className="block-3"></div>
            <div className="block-4b">
              <div className="block-4-floatbar"></div>
            </div>
          </div>

          <div>{children}</div>
          {footer}
        </div>
      </div>
      <ScrollToTopButton className="scroll-to-top-button" />
    </div>
  );
}

export default App;
