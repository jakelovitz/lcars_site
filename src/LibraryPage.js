import React, { useEffect, useState } from 'react';

function LibraryPage() {
  const [htmlContent, setHtmlContent] = useState('');

  useEffect(() => {
    fetch('/BookList.html')
      .then(response => response.text())
      .then(html => {
        // Parse the HTML string into a document object
        const parser = new DOMParser();
        const doc = parser.parseFromString(html, 'text/html');

        // Example: Remove all <script> tags from the document
        const scripts = doc.getElementsByTagName('script');
        while (scripts.length > 0) {
          scripts[0].parentNode.removeChild(scripts[0]);
        }

        // Example: Filter or manipulate other elements as needed

        // Serialize the document back to an HTML string
        const serializedHTML = doc.documentElement.outerHTML;

        // Set the filtered HTML content into the state
        setHtmlContent(serializedHTML);
      });
  }, []);

  return <div dangerouslySetInnerHTML={{ __html: htmlContent }} />;
}

export default LibraryPage;
