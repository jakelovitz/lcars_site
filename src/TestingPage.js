import React, { Component } from 'react';
import * as ReactDOM from "react-dom/client";
import logo from './logo.svg';
import './styling.css';

function TestingPage() {
    return (
        <div>Testing overview</div>
    )
}

export default TestingPage;
