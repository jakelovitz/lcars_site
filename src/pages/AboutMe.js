import React from 'react';
import './../assets/styling.css';
import favoriteThings from '../assets/favoriteThings';

class AboutMe extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            selectedButton: null
        };
    }

    handleButtonClick = (buttonName) => {
        this.setState({ selectedButton: buttonName });
    }

    componentDidMount() {
        this.setState({ selectedButton: null });
    }

    generateList = () => {
        const { selectedButton } = this.state;
        if (!selectedButton) return null;

        const thingData = favoriteThings.find(thing => thing.name === selectedButton);
        if (!thingData) return null;

        return (
            <div>
                <p>{thingData.blurb}</p>
                <ul className="lcars-list">
                    {thingData.items.map((item, index) => (
                        <li key={index}>{item}</li>
                    ))}
                </ul>
            </div>
        );
    }

    render() {
        return (
            <div >
                <div >
                    <div className="content">
                        <div className="flexbox">

                            <div className="col">
                                <img style={{width: "90%", height: "90%", objectFit: 'contain'}} src={process.env.PUBLIC_URL + '/pipes_alone.png'} alt="your humble web developer" />
                            </div>

                            <div className="col">
                                <p className="go-big">
                                    <span className="go-space-white">
                                My continuing mission: to pioneer unparalleled quality in digital products; to design and develop test suites comprehensive in their scope, depth, and supporting documentation; to boldly ensure consistency and confidence like never before.</span>
                                </p>
                                <p>I'm my mind, I'm a Code Integrity Officer, though in my near ten years of ensuring software products best meet the needs of its end users I've held such titles as Software Tester, Software (Development) Engineer in Test, and Quality Assurance Engineer.</p>
                                <p>Irrespective of the name of the role, I help software teams in their pursuit of that most elusive of goals, quality, through the noble and maligned art of testing.</p>
                            </div>
                        </div>

                        <div class="lcars-text-bar the-end">
                            <span>Off Duty</span>
                        </div>

                        <p>When not chasing down abstract notions in abstract places...something something</p>
                        <blockquote>
                            More and more I've come to think…that to call plays or stories or poems “good” or “bad” is often not very illuminating, whereas it can at times be extremely helpful to notice that “right now, when I read this particular poems, I feel well, I feel happy, I feel that I am getting something that I have needed”. It's not that these poems are “the best” poems or that they're “better” than certain others, but that for you, now, they are important and right.
                        </blockquote>
                        <p className='go-right'>Wallace Shawn</p>
                        <p>
                            I love how this quote lays bare how personal the relationship between a piece of art and the person experiencing it is. It's not to say that almost any object of art doesn't belong to a specific school which has it's own criteria for what might constite a 'good' work of art in said school.
                            Which also called to mind another saying I once heard nad has lived rent free in my head ever since
                        </p>
                        <blockquote>
                            No person ever steps in the same river twice, for it's not the same river and they're not the same person.
                        </blockquote>
                        <p className='go-right'>Heraclitus</p>

                        <div class="buttons">
                            <a onClick={() => this.handleButtonClick('Movies')}>Movies</a>
                            <a onClick={() => this.handleButtonClick('Albums')}>Albums</a>
                            <a onClick={() => this.handleButtonClick('Something')}>button 03</a>
                            <a onClick={() => this.handleButtonClick('Spaceships')}>Spaceships</a>
                            <a onClick={() => this.handleButtonClick('Books')}>Books</a>
                        </div>
                        {this.generateList()}
                    </div>
                </div>
            </div> 
        )
    }
}

export default AboutMe;