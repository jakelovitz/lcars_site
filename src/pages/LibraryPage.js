import React, { Component } from 'react';
import fetchAndParseBookList from '../utils/useFetchAndParseBookList';
import Book from './../components/Book'; // Adjust the import path as necessary

class LibraryPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      books: [],
      loading: true,
      error: null
    };
  }
  
  componentDidMount() {
    this.fetchBooks();
  }
  
  fetchBooks = () => {
    fetchAndParseBookList()
      .then(books => {
        this.setState({
          books: books,
          loading: false
        });
      })
      .catch(error => {
        this.setState({
          error: error,
          loading: false
        });
      }
    );
  }
  
  render() {
    const { loading, error, books } = this.state;
    const { onNavigate, footer } = this.props;


    if (loading) return <div>Loading books...</div>;
    if (error) return <div>Error loading books: {error.message}</div>;

    return (
      <div class="wrap" id="gap">
        <div class="left-frame">
          <div>
            <div class="panel-3"><span class="hop"></span></div>
            <div class="sidebar-buttons">
                  <a onClick={() => onNavigate(0)} id="0">About Me</a>
                  <a onClick={() => onNavigate(1)} id="1">Testing</a>
                  <a onClick={() => onNavigate(2)} id="2">Library</a>
            </div>
            <div class="panel-4"><span class="hop"></span></div>
                <div class="panel-5">Hello<span class="hop">Pipes</span></div>
                <div class="panel-6"><span class="hop"></span></div>
                <div class="panel-7"><span class="hop"></span></div>
                <div class="panel-8"><span class="hop"></span></div>
                <div class="panel-9">And<span class="hop">Again</span></div>
            </div>
            <div class="panel-10">Oh and<span class="hop">Yet Again</span></div>
          </div>
          <div class="right-frame">
            <div class="bar-panel">
            <div class="bar-6"></div>
            <div class="bar-7"></div>
            <div class="bar-8"></div>
            <div class="bar-9"><div class="bar-9-inside"></div></div>
          </div> 
          <div class="runner">
            <div class="corner-bg">
              <div class="corner"></div>
            </div>
            <div class="block-2"></div>
            <div class="block-3"></div>
            <div class="block-4b">
            <div class="block-4-floatbar"></div>
            </div>
          </div>
            {/* page content here */}
            <div class="content">
              <div class="flexbox">
                <h1>Library</h1>
                <div className="book-list">
                {books.map((book, index) => (
                    <Book key={index} {...book} />
                  ))}
                </div>
              <div/>
            </div>
            {footer}
          </div>
        </div>
      </div>
    );
  }
}
export default LibraryPage;