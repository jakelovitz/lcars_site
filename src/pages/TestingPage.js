import React from 'react';
import './../assets/styling.css';

function TestingPage() {
  return (
    <div>
      <div data-testid="testing-page-content">
        <div className="content">
          <div className="flexbox">
            <div className="col">
              <img
                style={{ width: '90%', height: '90%', objectFit: 'contain' }}
                src={process.env.PUBLIC_URL + '/glados.webp'}
                alt="glados"
              />
            </div>
            <div className="col">
              <p className="go-big">
                <span className="go-space-white">
                  "Since you went to all the trouble of waking me up, you must
                  really, really love to test." - GLaDOS
                </span>
              </p>
              <p>
                I really did love testing the Aperture Science Handheld Portal
                Device with GlaDOS circa 2008-2011. Nowadays (circa 2015 -
                present day), however, I've shifted to my focus from portals* to
                protocols**, primarily within that one final frontier of
                Software applications: the Electronic Health Record.
              </p>
              <p>
                *Not entirely true: I have tested a couple of patient portals
              </p>
              <p>**and APIs, databases, front ends, back ends, etc</p>
            </div>
          </div>
        </div>
        <div className="lcars-bar">
          <div className="lcars-bar-inner"></div>
        </div>

        <div className="content">
          <p>
            Had you told me that any time prior to 2015, I would have asked
            "...what precisely does 'testing software' entail? How is that a
            full time gig?"
          </p>
          <p>
            Here's how Glenford J. Myers, Tom Badgett, and Corey Sandler define
            the process of software testing in their in-depth and aptly named
            The Art of Software Testing, Third Edition:
          </p>
          <blockquote>
            Testing is the process of executing a program with the intent of
            finding errors
          </blockquote>
          <p>
            That's a quick and concise definition, which Myers et al. define on
            page 6 and spend most the rest of the book probing the implications
            of. In Cem Kaner, Jams Bach, and Bret Pettichord have a slightly
            different take, which they lay out in Indispensable Lessons Learned
            in Software Testing, a Context-Driven Approach:
          </p>
          <blockquote>Testing is done to find the right information</blockquote>
          <p>
            I like how Myers et al. describe the process, but prefer Kaner et
            al.'s description of our purpose. Thus my definition of (software)
            testing:
          </p>
          <blockquote>
            Testing is the process of executing a program with the intent of
            finding the right information.
          </blockquote>
          <p>
            Which is all very well and good, but still begs the question of how,
            exactly, one goes about doing that. Well, unlike the basics of the
            2008 video game Portal, and it's 2011 successor, it turns out that
            software testing is quite like software development - it's actually
            a pretty deep field
          </p>
        </div>
      </div>
    </div>
  );
}

export default TestingPage;
