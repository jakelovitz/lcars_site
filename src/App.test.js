import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';
import AboutMe from './pages/AboutMe';
import TestingPage from './pages/TestingPage';
import LibraryPage from './pages/LibraryPage';

test('renders About Me page', () => {
  render(
    <MemoryRouter initialEntries={['/']}>
      <App>
        <AboutMe />
      </App>
    </MemoryRouter>
  );
  expect(screen.getByText(/About Me/i)).toBeInTheDocument();
});

test('renders Testing page', () => {
  render(
    <MemoryRouter initialEntries={['/testing']}>
      <App>
        <TestingPage />
      </App>
    </MemoryRouter>
  );
  // Use data-testid to find the unique element
  expect(screen.getByTestId('testing-page-title')).toBeInTheDocument();
  expect(screen.getByTestId('testing-page-content')).toBeInTheDocument();
});

test('renders Library page', () => {
  render(
    <MemoryRouter initialEntries={['/library']}>
      <App>
        <LibraryPage />
      </App>
    </MemoryRouter>
  );
  expect(screen.getByText(/Library/i)).toBeInTheDocument();
});
