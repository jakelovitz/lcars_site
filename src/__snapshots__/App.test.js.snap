// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Header
    </div>
    <div
      class="wrap"
      id="gap"
    >
      <div
        class="left-frame"
      >
        <div>
          <div
            class="panel-3"
          >
            <span
              class="hop"
            />
          </div>
          <div
            class="sidebar-buttons"
          >
            <div
              href="/"
            >
              <div
                id="0"
              >
                About Me
              </div>
            </div>
            <div
              href="/testing"
            >
              <div
                id="1"
              >
                Testing
              </div>
            </div>
            <div
              href="/library"
            >
              <div
                id="2"
              >
                Library
              </div>
            </div>
          </div>
          <div
            class="panel-4"
          >
            <span
              class="hop"
            />
          </div>
          <div
            class="panel-5"
          >
            <span
              class="hop"
            />
          </div>
          <div
            class="panel-6"
          >
            <span
              class="hop"
            />
          </div>
          <div
            class="panel-7"
          >
            <span
              class="hop"
            />
          </div>
          <div
            class="panel-8"
          >
            <span
              class="hop"
            />
          </div>
          <div
            class="panel-9"
          >
            <span
              class="hop"
            />
          </div>
        </div>
        <div
          class="panel-10"
        >
          <span
            class="hop"
          />
        </div>
      </div>
      <div
        class="right-frame"
      >
        <div
          class="bar-panel"
        >
          <div
            class="bar-6"
          />
          <div
            class="bar-7"
          />
          <div
            class="bar-8"
          />
          <div
            class="bar-9"
          >
            <div
              class="bar-9-inside"
            />
          </div>
        </div>
        <div
          class="runner"
        >
          <div
            class="corner-bg"
          >
            <div
              class="corner"
            />
          </div>
          <div
            class="block-2"
          />
          <div
            class="block-3"
          />
          <div
            class="block-4b"
          >
            <div
              class="block-4-floatbar"
            />
          </div>
        </div>
        <div>
          <div>
            <div>
              <div
                class="content"
              >
                <div
                  class="flexbox"
                >
                  <div
                    class="col"
                  >
                    <img
                      alt="your humble web developer"
                      src="/pipes_alone.png"
                      style="width: 90%; height: 90%; object-fit: contain;"
                    />
                  </div>
                  <div
                    class="col"
                  >
                    <p
                      class="go-big"
                    >
                      <span
                        class="go-space-white"
                      >
                        My continuing mission: to pioneer unparalleled quality in digital products; to design and develop test suites comprehensive in their scope, depth, and supporting documentation; to boldly ensure consistency and confidence like never before.
                      </span>
                    </p>
                    <p>
                      I'm my mind, I'm a Code Integrity Officer, though in my near ten years of ensuring software products best meet the needs of its end users I've held such titles as Software Tester, Software (Development) Engineer in Test, and Quality Assurance Engineer.
                    </p>
                    <p>
                      Irrespective of the name of the role, I help software teams in their pursuit of that most elusive of goals, quality, through the noble and maligned art of testing.
                    </p>
                  </div>
                </div>
                <div
                  class="lcars-text-bar the-end"
                >
                  <span>
                    Off Duty
                  </span>
                </div>
                <p>
                  When not chasing down abstract notions in abstract places I try to keep things a little more real. To me, this primarily entails reading books, learning new things (or rather, trying to), and doing my damndest to live the good life. If you want to learn a more about what I'm all about I've taken the liberty of cataloguing some of my favorite things. I even riff a little bit on the 'why' behind these selections.
                </p>
                <div
                  class="buttons"
                >
                  <a>
                    Movies
                  </a>
                  <a>
                    Albums
                  </a>
                  <a>
                    Pok√©mon
                  </a>
                  <a>
                    Spaceships
                  </a>
                  <a>
                    Books
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div>
              <div
                class="content"
              >
                <div
                  class="flexbox"
                >
                  <div
                    class="col"
                  >
                    <img
                      alt="glados"
                      src="/glados.webp"
                      style="width: 90%; height: 90%; object-fit: contain;"
                    />
                  </div>
                  <div
                    class="col"
                  >
                    <p
                      class="go-big"
                    >
                      <span
                        class="go-space-white"
                      >
                        "Since you went to all the trouble of waking me up, you must really, really love to test." - GLaDOS
                      </span>
                    </p>
                    <p>
                      I really did love testing the Aperture Science Handheld Portal Device with GlaDOS circa 2008-2011. Nowadays (circa 2015 - present day), however, I've shifted to my focus from portals* to protocols**, primarily within that one final frontier of Software applications: the Electronic Health Record.
                    </p>
                    <p>
                      *Not entirely true: I have tested a couple of patient portals
                    </p>
                    <p>
                      **and APIs, databases, front ends, back ends, etc
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="lcars-bar"
              >
                <div
                  class="lcars-bar-inner"
                />
              </div>
              <div
                class="content"
              >
                <p>
                  Had you told me that any time prior to 2015, I would have asked "...what precisely does 'testing software' entail? How is that a full time gig?"
                </p>
                <p>
                  Here's how Glenford J. Myers, Tom Badgett, and Corey Sandler define the process of software testing in their in-depth and aptly named The Art of Software Testing, Third Edition:
                </p>
                <blockquote>
                  Testing is the process of executing a program with the intent of finding errors
                </blockquote>
                <p>
                  That's a quick and concise definition, which Myers et al. define on page 6 and spend most the rest of the book probing the implications of. In Cem Kaner, Jams Bach, and Bret Pettichord have a slightly different take, which they lay out in Indispensable Lessons Learned in Software Testing, a Context-Driven Approach:
                </p>
                <blockquote>
                  Testing is done to find the right information
                </blockquote>
                <p>
                  I like how Myers et al. describe the process, but prefer Kaner et al.'s description of our purpose. Thus my definition of (software) testing:
                </p>
                <blockquote>
                  Testing is the process of executing a program with the intent of finding the right information.
                </blockquote>
                <p>
                  Which is all very well and good, but still begs the question of how, exactly, one goes about doing that. Well, unlike the basics of the 2008 video game Portal, and it's 2011 successor, it turns out that software testing is quite like software development - it's actually a pretty deep field
                </p>
              </div>
            </div>
          </div>
          <div>
            Loading books...
          </div>
          <div>
            Footer
          </div>
        </div>
      </div>
    </div>
    <div
      class="wrap"
    />
  </div>
</DocumentFragment>
`;
